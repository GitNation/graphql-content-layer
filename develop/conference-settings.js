const fallback = require('./fallback-settings');

const confCodes = [
  'jsn',
  'gqconf',
  'mlconf',
  'qaconf',
  'doconf',
  'nodeconf',
  'rs',
];

const getSettings = () => {
  const settings = confCodes.reduce(
    (acc, code) => ({ ...acc, [code]: fallback }),
    {},
  );
  try {
    confCodes.forEach(code => {
      settings[
        code
      ] = require(`../../live-conferences/src/conferences/${code}/conference-settings`);
    });
    return settings;
  } catch (err) {
    if (err.code === 'MODULE_NOT_FOUND') {
      console.log(
        `\n\nCan't find live-conferences folder. Fallback settings will be used!\n\n`,
      );
      return settings;
    }
    throw err;
  }
};

module.exports = { getSettings };
