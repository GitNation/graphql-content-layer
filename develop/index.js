const fs = require('fs');
const path = require('path');

const { getContent } = require('../dist');
const { getSettings } = require('./conference-settings');

const CURRENT_CONF = 'rs';

const testLaunch = async () => {
  const settings = getSettings();
  const content = await getContent(settings[CURRENT_CONF]);

  fs.writeFileSync(
    path.resolve(__dirname, '../content-log.json'),
    JSON.stringify(content, null, 2),
  );
  // console.log('content', content);
};

testLaunch();
