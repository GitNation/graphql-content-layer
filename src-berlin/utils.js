const { labelColors } = require('./config');

const labelTag = prefix => label => {
  const colorInfo =
    labelColors.find(c => c.label === label) ||
    labelColors.find(c => c.label === null);
  return `${prefix}--${colorInfo.tag}`;
};

module.exports = {
  labelTag
};
