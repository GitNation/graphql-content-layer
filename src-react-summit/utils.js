const { markdownToHtml } = require('./markdown');

const prepareSpeakers = async speakers => {
  const allSpeakersAsync = speakers
    .filter(Boolean)
    .map(item => ({
      ...item.speaker,
      ...item,
      avatar: (item.speaker && item.speaker.avatar) || item.avatar || {},
    }))
    .map(
      async ({
        bio,
        githubUrl,
        twitterUrl,
        mediumUrl,
        ownSite,
        speaker,
        avatar,
        ...item
      }) => ({
        ...item,
        company: `${item.company}, ${item.country}`,
        photo: avatar && avatar.url,
        desc: await markdownToHtml(bio),
        github: githubUrl,
        twitter: twitterUrl,
        medium: mediumUrl,
        site: ownSite,
      })
    );
  const allSpeakers = await Promise.all(allSpeakersAsync)
  const uniqueSpeakers = new Set(allSpeakers.map(({ name }) => name));
  const readySpeakers = [...uniqueSpeakers].map(name => allSpeakers.find(speaker => speaker.name === name));
  return readySpeakers;
};

module.exports = {
  prepareSpeakers,
};
