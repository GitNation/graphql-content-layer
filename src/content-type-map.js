const contentTypeMap = {
  AdviceLounge: '31ae96063a25477f8f007379231c1482',
  ConferenceBrand: '2d89f139f3d042ab98d212062527eb82',
  ConferenceEvent: '20e4c3e5323f41cba1ae8840df6e6d32',
  DaySchedule: '2e92d35e0ac0421baf779aef56eec40d',
  Diversity: '55eb254b4ccb4cb1a4c380b7aaa7dc17',
  Extended: 'c20fac4698f94d148fcbb27989ef48e4',
  Faq: '4f4e1bf4ce884d6e99735af934488515',
  Job: '41e965599f914b84947ce04d901dc088',
  Mc: '29d69d9b48494ef38ac0d7a22012eeab',
  Page: '371641c28b044b00bf532b2dd6e1516d',
  PieceOfSpeakerInfo: 'd31070ab4611456a83fa837a6ccea527',
  PieceOfSponsorInfo: 'c52cf3fb6f1c41588ea7fc77a092f419',
  PieceOfText: 'c04bfcdd2eed47b880588f5506ed4f5a',
  Speaker: 'e981e312fd6c48a39230de9193be4db9',
  Sponsor: 'f188bcf25c8e419d9f8d10b0926dd825',
  // Talk: 'd5d4bfeb605c41569b30d478a03bbb0e',
  Track: '76d236bfe3344d0990028c5f69c69079',
  Workshop: '2ce17eb4016e4be5ae72bb0db484f3f0',

  // updated
  OrgEvent: 'cda52433c9fd4a6a818f71acd217e91d',
  Talk: '03b4712293184731b7d1bf074302f8e0',
  QA: '3473d8bc2712475c89e0b47d31b2ee97',
  LightningTalk: '09be1744ee4d465ea3f5a82972d2cb28',
  GroupLT: '694d3ca13ab44d45810c18fb01580c92',
  SpeakersRoom: 'b701d52be9d84599a3d46972acafc42f',
  DiscussionRoom: '07f1cfefde1942fbb63704917cbf2398',

};

module.exports = {
  contentTypeMap,
};
