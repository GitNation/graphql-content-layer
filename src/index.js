const { GraphQLClient } = require('graphql-request');

const { credentials, conferenceTitle, eventYear } = require('./config');
const textContent = require('./fetch-texts');
const pageContent = require('./fetch-pages');
const brandContent = require('./fetch-brand');
const speakerContent = require('./fetch-speakers');
const performanceContent = require('./fetch-performance');
const sponsorContent = require('./fetch-sponsors');
const talksContent = require('./fetch-talks');
const workshopContent = require('./fetch-workshops');
const mcContent = require('./fetch-mc');
const faqContent = require('./fetch-faq');
const extContent = require('./fetch-extended');
const jobsContent = require('./fetch-jobs');
const committeeContent = require('./fetch-committee');

const createClient = ({ endpoint, token }) => {
  return new GraphQLClient(endpoint, {
    headers: {
      authorization: `Bearer ${token}`,
    },
  });
};

const client = createClient(credentials);
const queriesData = [];

const getQueriesData = content => {
  try {
    const { fetchData, ...data } = content;
    if (!Object.keys(data).length) return;
    queriesData.push(data);
  } catch (err) {}
};

const getContent = async () => {
  const fetchAll = [
    textContent,
    pageContent,
    brandContent,
    speakerContent,
    performanceContent,
    sponsorContent,
    talksContent,
    workshopContent,
    mcContent,
    faqContent,
    extContent,
    jobsContent,
    committeeContent,
  ].map(async content => {
    try {
      getQueriesData(content);
      return await content.fetchData(client, { conferenceTitle, eventYear });
    } catch (err) {
      console.error(err);
    }
  });

  const contentArray = await Promise.all(fetchAll);
  const contentMap = contentArray.reduce((content, piece) => {
    try {
      const newKeys = Object.keys(piece);
      const existentKeys = Object.keys(content);
      const intersectedKeys = newKeys.filter(k => existentKeys.includes(k));
      intersectedKeys.forEach(k => {
        piece[k] = { ...content[k], ...piece[k] };
      });
    } catch (err) {}
    return { ...content, ...piece };
  }, {});
  return contentMap;
};

module.exports = {
  getContent,
  queriesData,
};
