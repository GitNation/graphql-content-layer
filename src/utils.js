const { markdownToHtml } = require('./markdown');

const tagColors = {
  'NodeJS': {
    tagBG: '#7AB464',
    color: '#fff',
  },
  'WebGL': {
    tagBG: '#ff7302',
    color: '#ffffff'
  },
  'Ecosystem': {
    tagBG: '#fff40d',
    color: '#ae4f01'
  },
  'Language': {
    tagBG: '#fff40d',
    color: '#ae4f01'
  },
  'GraphQL': {
    tagBG: '#f200fa',
    color: '#400042',
  },
  'VueJS': {
    tagBG: '#4EBA87',
    color: '#fff',
  },
  'Performance': {
    tagBG: '#00ed24',
    color: '#00410a',
  },
  'TypeScript': {
    tagBG: '#61DAFB',
    color: '#030303'
  },
  default: {
    tagBG: 'black',
    color: 'white'
  }
};

const getSocials = speaker => {
  const ICONS = {
    githubUrl: 'gh',
    twitterUrl: 'tw',
    mediumUrl: 'med',
    ownSite: 'site',
  };
  const { githubUrl, twitterUrl, mediumUrl, ownSite, companySite } = speaker;
  const socials = Object.entries({ githubUrl, twitterUrl, mediumUrl, ownSite, companySite })
    .map(([key, val]) => (val && { link: val, icon: ICONS[key] }))
    .filter(Boolean);
  return socials;
};

const getLabelColor = label => {
  const colors = tagColors[label] || tagColors.default;
  return colors;
};

const prepareSpeakers = speakers => speakers
  .map(item => ({
    ...item.speaker,
    ...item,
    avatar: item.speaker.avatar || {},
  }))
  .map(
    async ({
      bio,
      speaker,
      avatar,
      ...item
    }) => ({
      ...item,
      company: `${item.company}, ${item.country}`,
      avatar: avatar.url,
      bio: await markdownToHtml(bio),
      socials: getSocials(item),
      ...getLabelColor(item.label),
    })
  );

module.exports = {
  getLabelColor,
  prepareSpeakers,
  tagColors,
};
