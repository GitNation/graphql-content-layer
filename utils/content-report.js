import hash from 'object-hash';

const createPagesReport = pages => {
  const pagesKeys = Object.keys(pages);
  const report = pagesKeys.map(key => {
    const page = pages[key];
    const {
      id,
      titleSeo,
      description,
      seoDescription,
      titlePage,
      ...rest
    } = page;
    return {
      pageCode: key,
      id,
      titleSeo,
      description,
      seoDescription,
      titlePage,
      content: hash(rest),
    };
  });
  return report;
};

export const createReport = content => {
  try {
    const {
      pages,
      conference,
      speakers,
      sponsors,
      schedule,
      tracks,
      talks,
      workshops,
      ...rest
    } = content;
    return {
      pages: createPagesReport(pages),
      conference,
    };
  } catch (err) {
    console.error(err);
    return {};
  }
};
