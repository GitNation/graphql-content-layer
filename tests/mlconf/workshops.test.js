import { getSettings } from '../../develop/conference-settings';
import { getContent } from '../../src';
import { createReport } from '../../utils/content-report';

describe('RS', () => {
  const settings = getSettings();

  it('should be correct workshops data', async () => {
    const content = await getContent(settings.mlconf);
    const report = createReport(content);
    expect(report.workshops).toMatchSnapshot();
  });
});
