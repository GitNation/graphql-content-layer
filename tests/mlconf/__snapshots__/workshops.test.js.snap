// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RS should be correct workshops data 1`] = `Array []`;
